src/pkg/cli/configResolution.go

```diff
@@ -0,0 +1,104 @@
+package cli
+
+import (
+	"slices"
+	"sort"
+
+	"github.com/DefangLabs/defang/src/pkg/cli/compose"
+	"github.com/DefangLabs/defang/src/pkg/term"
+	"github.com/compose-spec/compose-go/v2/types"
+)
+
+type configOutput struct {
+	Service string `json:"service"`
+	Name    string `json:"name"`
+	Value   string `json:"value,omitempty"`
+	Source  Source `json:"source,omitempty"`
+}
+
+const configMaskedValue = "*****"
+
+type Source int
+
+const (
+	SourceUnknown Source = iota
+	SourceComposeFile
+	SourceDefangConfig
+	SourceDefangAndComposeFile
```

Comment:
It's confusing to say that a value comes from two sources. I know you want to use this value for interpolated env, but I'd just call it "interpolation" then, and show the vars used in the interpolation as their own separate rows in the output.